poly-synth/src/poly-synth.js

```diff
@@ -32,15 +32,45 @@ class PolySynth {
     this.releasedVoices_ = [];
 
     // The maximum and minimum cutoffs are experimentally determined.
-    this.lowPassMaxCutoff = 16000;
-    this.lowPassMinCutoff = 60;
+    this.maxCutoff = 16000;
+    this.minCutoff = 60;
 
+    // The maximum and minimum ADSR values refer to the ADSR for the gain as
+    // well as the ADSR for the filter. These values are experimentally
+    // determined.
+    this.maxAttack = 10;
+    this.minAttack = 0;
+    this.minSustain = 0.1;
+    this.maxSustain = 1;
+
+    // The minimum decay is > 0 since there will be a conflict if the time to
+    // the sustain and time to maximum level are identical.
+    this.minDecay = 0.01;
+    this.maxDecay = 10;
+    this.minRelease = 0;
+    this.maxRelease = 10;
+
+    // A detune amount of 1 will correspond to 2400 cents detuning from the
+    // cutoff value.
+    this.minDetuneAmount = 0;
+    this.maxDetuneAmount = 5;
+
+    this.minQ = 0;
+    this.maxQ = 100;
+
+    // The initial values for the parameters are experimentally determined.
     this.parameters_ = {
-      cutoff: this.lowPassMaxCutoff,
-      attack: 0,
-      decay: 0,
-      sustain: 0.1,
-      release: 0
+      gainAttack: 0,
+      gainDecay: this.minDecay,
+      gainSustain: this.minSustain,
```

Comment:
Is minSustain (0.1?) really the right value for the default?  Pretty drastic change in volume from attack to sustain.